service/history/transferQueueTaskExecutorBase.go

```diff
@@ -150,6 +152,7 @@ func (t *transferQueueTaskExecutorBase) pushWorkflowTask(
 		TaskQueue:              taskqueue,
 		ScheduleId:             task.ScheduleID,
 		ScheduleToStartTimeout: workflowTaskScheduleToStartTimeout,
+		Clock:                  vclock.NewShardClock(t.shard.GetShardID(), task.TaskID),
```

Comment:
@yiminc e.g. logic is using the task ID (generated from the sequence ID) as shard clock stamp